P5/Source/Specs/ruby.xml

```diff
@@ -0,0 +1,57 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+    Copyright TEI Consortium.
+    Dual-licensed under CC-by and BSD2 licences
+    See the file COPYING.txt for details
+    $Date$
+    $Id$
+-->
+<?xml-model href="http://jenkins.tei-c.org/job/TEIP5-dev/lastSuccessfulBuild/artifact/P5/release/xml/tei/odd/p5.nvdl" type="application/xml" schematypens="http://purl.oclc.org/dsdl/nvdl/ns/structure/1.0"?>
+<elementSpec xmlns="http://www.tei-c.org/ns/1.0" ident="ruby" module="core">
+<gloss versionDate="2021-01-30">ruby container</gloss>
+<gloss versionDate="2021-01-31" xml:lang="ja">ルビのためのコンテナ要素。</gloss>
+<desc versionDate="2020-02-28" xml:lang="en">contains a
+  passage of base text along with its associated ruby gloss(es).</desc>
+<desc versionDate="2021-01-31" xml:lang="ja">ルビ及びその対象となるテキストを含む。</desc>
+<classes>
+  <memberOf key="att.global"/>
+  <memberOf key="att.typed"/>
+  <memberOf key="model.phrase"/>
+</classes>
+  <content>
```

Comment:
I am feeling a bit sheepish about this, as I think it is too late to do anything now. But this content model is clearly too restrictive. Elements like `<addSpan>`, `<certainty>`, and `<metamark>` (all members of model.global) surely should be allowed inside `<ruby>`.

I am also thinking maybe `<ruby>` should be a member of att.breaking.

I do not think either of these should be addressed now. Separate tickets for 4.x.0 (where x ≥ 3), I guess.

Reply:
This is a bit worrying. The complex stuff should really appear in rb rather than in ruby, no? Or am I misunderstanding?

Reply:
But what if it is not actually part of the base text, but rather a mark (read `<metamark>`) on the manuscript that indicates which ruby annotation should be associated with which ruby base character? Or you want to use the `<certainty>` element to express uncertainty about a ruby interpretation? Or the added span of text is (perhaps incorrectly or inappropriately) inserted in between the ruby base and ruby annotation?

All of these are quite reasonable, but also things that @hcayless would say are not (yet) needed as no one has asked for them (yet).

Reply:
I think some of these projected issues might be based on a misunderstanding, and without specific examples I don't know how to respond to them. We know that after this release there will be at least seven tickets arising out of @duncdrum's examples that will require either enhanced examples/recommendations and/or schema adjustments, and the certainly issue perhaps makes an eighth (although I think what you are expressing uncertainty about in most cases is either going to be something in the rb or the rt). There will also be more from the Japanese community as they develop their own guidelines based on this release. But without genuine examples it's difficult to imagine what needs to be done. The only question that matters, I think, is whether at this point we are building any decision into the Guidelines which we later have to roll back in such a way as to break backwards compatibility; an overly-restrictive content model is not such a problem, where the reverse would be. 
Regarding the question of uncertainty about which rb/rt relationship is correct, a simple choice element is an obvious solution, or the use of the `@select` attribute.

Reply:
> an overly-restrictive content model is not such a problem, where the reverse would be

Agreed, completely. Which is why I am not suggesting any changes at this time.

Reply:
My first job as soon as the merge is done is going to be to go through (with your help, I hope) all @duncdrum's tickets on the rubyForTEI repo and turn them into appropriate issues on the TEI repo, tagged for a future milestone. We should also refine these concerns into specific tickets, hopefully with real examples from the scholars involved.